akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala

```diff
@@ -329,9 +332,13 @@ private[http2] trait Http2StreamHandling { self: GraphStageLogic with LogHelper
 
   case class OpenSendingData(outStream: OutStream, correlationAttributes: Map[AttributeKey[_], _]) extends StreamState with Sending {
     override def handle(event: StreamFrameEvent): StreamState = event match {
-      case _: ParsedHeadersFrame => expectIncomingStream(event, HalfClosedRemoteSendingData(outStream), Open(_, outStream), correlationAttributes)
-      case w: WindowUpdateFrame  => handleWindowUpdate(w)
-      case _                     => receivedUnexpectedFrame(event)
+      case _: ParsedHeadersFrame =>
+        expectIncomingStream(event, Some(outStream), HalfClosedRemoteSendingData(outStream), Open(_, outStream), correlationAttributes)
+      case w: WindowUpdateFrame =>
+        handleWindowUpdate(w)
+      case _ =>
+        outStream.cancelStream()
```

Comment:
`receivedUnexpectedFrame` tries to push GOAWAY, but that is for entire connection so no problem with canceling the outstream first right?